state_base.py

```diff
@@ -28,16 +29,24 @@ class StateBase(Persistence, GroupBy, Block):
     def __init__(self):
         super().__init__()
         self._initial_state = None
+        self._changed_state = None
         self._state_locks = defaultdict(Lock)
         self._safe_lock = Lock()
         self._states = {}
+        self._states_to_persist = {}
 
     def persisted_values(self):
         """Persist states using block mixin."""
-        return ["_states"]
+        if self._changed_state == False:
+            return []
+        else:
+            return ['_states_to_persist']
```

Comment:
Let's make this change backwards compatible by still persisting `_states` and instead adding something like `_states_not_to_persist`? Probably can come up with a better name that that. Maybe `_states_without_initial`?